src/Raven.Server/Documents/PeriodicBackup/GoogleCloudStorage/RavenGoogleCloudStorageClient.cs

```diff
@@ -0,0 +1,158 @@
+                            break;
+                        case UploadStatus.Completed:
+                            _progress.UploadProgress.ChangeState(UploadState.Done);
+                            break;
+                        case UploadStatus.Uploading:
+                            _progress.UploadProgress.ChangeState(UploadState.Uploading);
+                            break;
+                    }
+
+                    _progress.UploadProgress.UploadedInBytes = p.BytesSent;
+                    _progress.OnUploadProgress();
+                }));
+        }
+
+        public async Task DownloadObjectAsync(string fileName, Stream stream)
+        {
+            await _client.DownloadObjectAsync(
+                _bucketName,
+                fileName,
+                cancellationToken: CancellationToken,
+                destination: stream
+            ).ConfigureAwait(false);
+        }
+
+        public async Task<Object> GetObjectAsync(string fileName)
+        {
+            return await _client.GetObjectAsync(
+                _bucketName,
+                fileName,
+                cancellationToken: CancellationToken
+            ).ConfigureAwait(false);
+        }
+
+        public async Task DeleteObjectAsync(string fileName)
+        {
+            await _client.DeleteObjectAsync(
+                _bucketName,
+                fileName,
+                null,
+                CancellationToken
+            ).ConfigureAwait(false);
+        }
+
+        public PagedEnumerable<Buckets, Bucket> ListBuckets()
+        {
+            if (_projectId == null)
+                throw new ArgumentException($"Project Id was not found in Google credentials json, " +
+                                            $"please make sure that '{ProjectIdPropertyName}' is part of the credentials");
+
+            return _client.ListBuckets(_projectId);
+        }
+
+        public PagedAsyncEnumerable<Objects, Object> ListObjectsAsync()
+        {
+            return _client.ListObjectsAsync(_bucketName);
+        }
+
+        public async Task TestConnection()
+        {
+            try
+            {
+                await _client.GetBucketAsync(_bucketName, cancellationToken: CancellationToken);
+
+                if (await _client.TestBucketIamPermissionsAsync(_bucketName, new[] {"storage.objects.create"}, cancellationToken: CancellationToken) == null)
+                {
+                    throw new InvalidOperationException(
+                        $"Can't create an object in bucket '{_bucketName}', " +
+                        $"The permission 'storage.objects.create' is missing");
+                }
+            }
+            catch (Google.GoogleApiException e)
+                when (e.Error.Code == 403)
+            {
+                throw new InvalidOperationException($"Google credentials json does not have access to project {_projectId ?? "N/A"}");
```

Comment:
You need to include the actual exception in _all_ the exceptions you rethrow

Reply:
Fixed